gator/files.py

```diff
@@ -23,10 +23,45 @@ def create_path(*args, file, home):
     return final_path
 
 
-def check_file_in_directory(*args, file, home):
-    """Returns true if the specified file is in the directory"""
+def case_insensitive_check_file_in_directory(*args, file, home):
+    """Returns true if the case-insensitive specified file is in the directory"""
     # create the path so that it has this structure:
     # <home> + <any paths in *args, without their anchor> + <file>
     file_for_checking_path = create_path(*args, file=file, home=home)
     # return true if the specified file exists
     return file_for_checking_path.is_file()
+
+
+def case_sensitive_check_file_in_directory(*args, file, home):
+    """Returns true if the case-sensitive specified file is in the directory"""
+    # create the path so that it has this structure:
+    # <home> + <any paths in *args, without their anchor> + <file>
+    file_for_checking_path = create_path(*args, file=file, home=home)
+    # get parent, i.e., the containing directory for the specified file
+    file_parent = file_for_checking_path.parent
+    # create a generator of all of the files in the parent directory
+    file_parent_glob = file_parent.glob("**/*")
+    # assume that the file with the correct name has not been found
+    # and prove otherwise by iterating through the generator of files
+    file_found = False
+    for current_file in file_parent_glob:
+        # the case-sensitive file name has been found
+        if str(current_file.name) == file:
```

Comment:
I may just not quite understand the logic here, but doesn't `**/*` expand files recursively? If that is the case, then this would look for `file` in every directory below `home`, not just `home`.

For example, a structure like below would pass for a call such as `case_sensitive_check_file_in_directory(file="reflection.md", "writing")`, when it should probably not.

```
- repo
  - writing
    - refraction.md
    - stuff
      - reflection.md
```

Reply:
Yes, @Michionlion, your point is correct. I think that it should only be the glob containing `*`, right?

Reply:
Yes, that makes sense; this may also be a problem however, because at least by default `*` does not include hidden files such as `.travis.yml`.

Reply:
Yes, agreed. I have merged your changes to this PR, fixed another misleading docstring that your commits did not fix, and then added `*.*`, which should handle your dotfile concern, right?

Reply:
I'm not sure, let me test it.

Reply:
Okay, `*.*` does indeed work, however it doesn't capture directories, which I don't think matters in this case. However, `*` works as well, at least in pathlib on MacOS, and also capturing directories. I think the current value is fine, but we should remember this in case there is any weirdness in the future.

Reply:
That sounds fine. It is worth noting that a future version will not be using the constant variable that I defined in this module and instead use a variable in `constants`. I will ensure that there is a comment in that file by that declaration that explains the issue. Let me know if there are other steps that you would like me to take, okay?

Reply:
Nothing else comes to mind! I'm merging this as soon as everything passes.